### ISSUE TITLE
Use terraform-docs native saving to file

### ISSUE THREAD

**Issue by khos2ow** [antonbabenko/pre-commit-terraform#179]
In terraform-docs `v0.12.0`, we've added ability to insert generated output directly into a target file (https://github.com/terraform-docs/terraform-docs/issues/286)! This can replace all the sed/awk/perl and drastically simplify the process to the point of only executing:

```console
terraform-docs --output-file README.md --output-mode inject .
```

Check out all the available option to [configure](https://terraform-docs.io/user-guide/configuration/#output) this.

We've just cut a [`v0.12.0-beta.1`](https://github.com/terraform-docs/terraform-docs/releases/tag/v0.12.0-beta.1) for folks to start experimenting. Please feel free to start using and file any issues or feedback.

**Comment by jonmaestas** [comment]
Does it make sense to deprecate `terraform_docs_without_aggregate_type_defaults` and `terraform_docs_replace`? 

Why not just support **native** `terraform-docs` usage similar to how `tflint` is setup? It seems like the other `terraform_docs_*` options were there because of the lack of features out-of-the-box with `terraform-docs` at the time, and that's no longer the case.

**Comment by kevcube** [comment]
This fix would be very helpful to me! 😸

**Comment by dzavalkin-scayle** [comment]
Any chance to get this fixed and remove all those crazy dependencies like perl?
`terraform-docs` is already at version 16 (for a long time btw) and with correct `.terraform-docs.yml` I'm ending up with my README.md corrupted by the hook:

```md
<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
README.md updated successfully
<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```

While when I run `terraform-docs` from cli README.md is updated as expected.
Basically, `terraform-docs` is currently unusable if we want to preserve possibility to run `terraform-docs` manually from CLI...

Just for the reference, my `.terraform-docs.yml` looks like this:

```yaml
formatter: "markdown table"
output:
  file: README.md
  mode: inject
  template: |-
    <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
    {{ .Content }}
    <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```

Would be also great to get rid of custom `<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->` and `<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->` markers and use [default](https://terraform-docs.io/user-guide/configuration/output/) `<!-- BEGIN_TF_DOCS -->` and `<!-- END_TF_DOCS -->`...

**Comment by gchappel** [comment]
> Any chance to get this fixed and remove all those crazy dependencies like perl? `terraform-docs` is already at version 16 (for a long time btw) and with correct `.terraform-docs.yml` I'm ending up with my README.md corrupted by the hook:
> 
> ```
> 
> README.md updated successfully
> 
> ```
> 
> While when I run `terraform-docs` from cli README.md is updated as expected. Basically, `terraform-docs` is currently unusable if we want to preserve possibility to run `terraform-docs` manually from CLI...
> 
> Just for the reference, my `.terraform-docs.yml` looks like this:
> 
> ```yaml
> formatter: "markdown table"
> output:
>   file: README.md
>   mode: inject
>   template: |-
>     
>     {{ .Content }}
>     
> ```
> 
> Would be also great to get rid of custom `<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->` and `<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->` markers and use [default](https://terraform-docs.io/user-guide/configuration/output/) `<!-- BEGIN_TF_DOCS -->` and `<!-- END_TF_DOCS -->`...

@dzavalkin-scayle I'm coming across this behaviour too - there's some kind of conflict between `terraform-docs` which now writes to the `README.md` natively instead of printing the rendered text to `stdout` and the current `pre-commit` hook which takes `stdout` and does its own replacement into the `README.md`

I've worked around this by adding an argument into my hook which restores the previous behaviour of making `terraform-docs` return the parsed output on `stdout`
```
        args:
          - --args=--output-file=""
```

Maybe this helps you too? In the long term I agree that the entire wrapper should just be simplified and require a recent version of `terraform-docs` to do this natively; but I don't have time for that at the moment so this was a quick band-aid for me.

**Comment by MaxymVlasov** [comment]
@dzavalkin-scayle @gchappel 
* `--output-file=""` set by default starting from https://github.com/antonbabenko/pre-commit-terraform/releases/tag/v1.92.2
* Starting from https://github.com/antonbabenko/pre-commit-terraform/releases/tag/v1.93.0 `<!-- BEGIN_TF_DOCS -->` and `<!-- END_TF_DOCS -->` markers used by default (with under-the-hood replacement of old-one markers).
    * Previously, you just can specify `- --hook-config=--use-standard-markers=true` (starting from https://github.com/antonbabenko/pre-commit-terraform/releases/tag/v1.86.0)

Please let me know if 1.93.0 can be perceived as breaking change for your workaround with
```yaml
  template: |-
    <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
    {{ .Content }}
    <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```

We can add replacement of `<!-- -->`  in specified `--args=--config=` file to hook, but I am not sure is it worth it, as we never promised to support such workarounds. 

I see that it [used in 32 public repos](https://github.com/search?q=%22%3C%21--+BEGINNING+OF+PRE-COMMIT-TERRAFORM+DOCS+HOOK+--%3E%22+%28path%3A*.yaml+OR+path%3A*.yml%29&type=Code&ref=advsearch&l=&l=&p=1)

With 1 quite problematic case like in this TFLint [Azure CI](https://github.com/MattMencel/azdo-pipeline-examples/blob/72ffd74f7f305e73b0c942bc01a757b8464db93f/Terraform/Terraform-Plan-Readiness.yml#L71-L93) setup:

Is pointing all folks which use such workarounds to

https://github.com/antonbabenko/pre-commit-terraform/blob/99fceb87b07cdbb97111c3b4eb17d3f70d36e13c/README.md#L584-L590

will be enough? 

---

Also, changes in v1.93.0 open the path to implementation of  `Use terraform-docs native saving to file` without needs of breaking changes, as used markers now are the same

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.94.0 :tada:
